### ISSUE TITLE
Cannot import QML from QM On Demand 3 2018.10

### ISSUE THREAD

**Issue by ruudhabets** [lucisgit/moodle-qformat_qml#5]
When an QML file, exported from QM On Demand 2018.10, is imported. this error occurs:

The type of the file you selected (text/xml) does not match the type expected by this import format (application/xml).

the qml is attached

**Comment by ruudhabets** [comment]
please rename when testing

[structuur voor demonstratie.qml.txt](https://github.com/lucisgit/moodle-qformat_qml/files/2571816/structuur.voor.demonstratie.qml.txt)

**Comment by tonyjbutler** [comment]
Thanks Ruud. I can't reproduce your error (although I do get a different one relating to unsupported question types, but we need to deal with this one first). I think it may have something to do with the recent changes to the way Moodle detects and handles file types.
Could you try adding a custom file type at /admin/tool/filetypes/index.php please? Configure it as per the attached screenshot then attempt your import again and let me know how it goes.
![qml](https://user-images.githubusercontent.com/1395170/48357758-39d97880-e691-11e8-9236-985cdbaaf420.png)
Thanks,
Tony

**Comment by ruudhabets** [comment]
Thanks tony.
I will have to contact our hosting company for that.

I will let you know what the results are.

Ruuf



Verzonden vanaf mijn Samsung Galaxy-smartphone.


-------- Oorspronkelijk bericht --------
Van: Tony Butler <notifications@github.com>
Datum: 12-11-18 16:40 (GMT+01:00)
Aan: lucisgit/moodle-qformat_qml <moodle-qformat_qml@noreply.github.com>
Cc: "Habets, WGR (Ruud)" <ruud.habets@zuyd.nl>, Author <author@noreply.github.com>
Onderwerp: Re: [lucisgit/moodle-qformat_qml] Cannot import QML from QM On Demand 3 2018.10 (#5)


Thanks Ruud. I can't reproduce your error (although I do get a different one relating to unsupported question types, but we need to deal with this one first). I think it may have something to do with the recent changes to the way Moodle detects and handles file types.
Could you try adding a custom file type at /admin/tool/filetypes/index.php please? Configure it as per the attached screenshot then attempt your import again and let me know how it goes.
[qml]<https://user-images.githubusercontent.com/1395170/48357758-39d97880-e691-11e8-9236-985cdbaaf420.png>
Thanks,
Tony

**Comment by ruudhabets** [comment]
nah,
didn't need to hassle the file system, I got a little confused there. 
I changed the admin settings and 'voila' the file imported nicely.
Yes there were errors but they disappeared when I imported a file that only consisted of supported questiontypes. That is something my instructors can worry about.
Many thanks, here at Zuyd University. 
When you intend to expand the supported questiontypes I would be happy to test things for you.
